parser/result.go

```diff
@@ -475,14 +476,30 @@ func (r *result) asEnumReservedRange(rng *ast.RangeNode, handler *reporter.Handl
 	return rr
 }
 
-func (r *result) asMessageDescriptor(node *ast.MessageNode, isProto3 bool, handler *reporter.Handler) *descriptorpb.DescriptorProto {
+func (r *result) asMessageDescriptor(node *ast.MessageNode, isProto3 bool, handler *reporter.Handler, depth int) *descriptorpb.DescriptorProto {
 	msgd := &descriptorpb.DescriptorProto{Name: proto.String(node.Name.Val)}
 	r.putMessageNode(msgd, node)
-	r.addMessageBody(msgd, &node.MessageBody, isProto3, handler)
+	// don't bother processing body if we've exceeded depth
+	if r.checkDepth(depth, node, handler) {
+		r.addMessageBody(msgd, &node.MessageBody, isProto3, handler, depth)
+	}
 	return msgd
 }
 
-func (r *result) addMessageBody(msgd *descriptorpb.DescriptorProto, body *ast.MessageBody, isProto3 bool, handler *reporter.Handler) {
+func (r *result) checkDepth(depth int, node ast.MessageDeclNode, handler *reporter.Handler) bool {
+	if depth != 32 {
```

Comment:
Should this be < for safety? I'm assuming that might only occur if we miss a call to checkDepth while recursing?